src/Serializer/AbstractItemNormalizer.php

```diff
@@ -313,9 +320,9 @@ protected function denormalizeRelation(string $attributeName, PropertyMetadata $
 
         if (!\is_array($value)) {
             // repeat the code so that IRIs keep working with the json format
-            if (true === $this->allowPlainIdentifiers && $this->itemDataProvider) {
+            if (true === $this->allowPlainIdentifiers && ($this->iriConverter instanceof IriToIdentifierConverterInterface && $this->iriConverter instanceof IriConverterInterface)) {
```

Comment:
Shouldn't we support the other path? I mean what if IRIConverter is not an instance of `IriToIdentifierConverterInterface` (case where someone extends the class)?